### PR TITLE
Implement scholarship filters (#630, #633, #636, #637, #646)

### PR DESCRIPTION
**Pull request by blms**
## visual review notes

Percy: [light mode](https://percy.io/2cf28a24/geniza/builds/16000192/changed/901901801) | [dark mode](https://percy.io/2cf28a24/geniza/builds/16000192/changed/901902054)

In scope:
- filter dialog, look/placement of the filters within it
- look/placement of "Apply" button
- placement of the "Filters" button in the [Document Search page](https://percy.io/2cf28a24/geniza/builds/15963964/changed/899951378)
- all of the above, mobile and desktop

Not in scope:
- the safari-only bug for the document type filter
- the partial search page that appears from underneath the filter dialog (it's a percy bug/quirk)
- fonts, horizontal spacing

## What this PR does

- Per #640:
  - solr boolean indices for digital editions, translations, and discussions
  - new Django form classes for boolean facet checkboxes (with some of the shared features moved into a reusable mixin)
- Per #633:
  - UI: filter button, filters modal, and styles for the new filter types
- Per #646:
  - "Apply" button UI and functionality

### CODE REVIEW

File: geniza/corpus/views.py [hunk 1]
```diff
@@ -95,6 +95,9 @@ def get_queryset(self):
             DocumentSolrQuerySet()
             .filter(status=Document.PUBLIC_LABEL)
             .facet_field("type", exclude="type", sort="value")
+            .facet_field("has_digital_edition")
+            .facet_field("has_translation")
+            .facet_field("has_discussion")
```
Comment (rlskoeser) [inline geniza/corpus/views.py:100]: these facets can be added in a single call with `facet` if we don't need extra options like the exclusion and sort we have on type
```suggestion
           .facet("has_digital_edition", "has_translation", "has_discussion")
```

**Review by rlskoeser — APPROVED** [review]
Looks good, working well.

The only thing that was confusing to me in the UI is when the new filters have zero counts — at first this was because I failed to reindex (my fault), but there are still scenarios where you could get a result with zero documents in some or all of these categories. I think it's confusing to have no number and no checkmark; maybe this is a more of a design question — is this a case that Gissoo accounted for in the various states for this component? (I'm not finding it). Can we show the zero but disable so the filter can't be selected?

**Comment by blms** [comment]
@rlskoeser Regarding disabling fields when the count is zero: it works, except that it doesn't let you remove filters when they are causing the count to be zero. For example, if you query "document" and turn on all 3 of the new filters, then you change your query to "documentary" while leaving the 3 filters applied, it's impossible to unapply them without restarting the search. A good argument for a "reset filters" option—but at the moment it causes problems. Should I leave them enabled but just show the number 0?

**Comment by blms** [comment]
For future reference, since it's got a few moving parts, if we ever do implement that:

`populate_from_facets`:
```python
if count == 0:
    self.disabled = True
```

template:
```django
<label for="{{ form.has_transcription.auto_id }}" {% if form.has_transcription.field.disabled %}disabled{% endif %}>
```

css:
```css
&[disabled] {
    pointer-events: none;
    cursor: default;
    color: var(--disabled);
    .count::after {
        color: var(--disabled);
    }
}
```

**Comment by blms** [comment]
Seems that it was required to put that `facet` call first before `facet_field`.

Also went ahead and added the apply button—the only tweak this required was disabling the submit debounce. Otherwise sometimes the `submit()` action wasn't called.

File: geniza/corpus/templates/corpus/document_list.html [hunk 2]
```diff
@@ -43,7 +43,7 @@ <h1 class="sr-only">{{ page_title }}</h1>
         </fieldset>
 
         <fieldset id="filters" aria-expanded="false">
-            <a href="#" role="button" id="close-filters-modal">
+            <a href="#" role="button" id="close-filters-modal" data-search-target="closeModal">
```
Comment (rlskoeser) [inline geniza/corpus/templates/corpus/document_list.html:46]: Can we name `closeModalFilters` (`closeFilterModal`? or even just `closeFilters`? )? Would like a little more context so this is more immediately obvious; and it's possible we'll have other modals.

**Review by rlskoeser — COMMENTED** [review]
When I apply the filters, it submits the form but it reloads the page with the filter modal open.

**Comment by gissoo** [comment]
@blms thank you!!! These look great. I think the only thing that does not match the designs is that when a filter is active (expanded) its drop shadow is green on light mode and pink in dark mode. If we can't do it that's fine, not a deal breaker, it just looks better.

File: sitemedia/js/controllers/search_controller.js [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
     connect() {
-        useDebounce(this);
+        // useDebounce(this);
     }
 
     submit() {
+        // Close modal if open before submitting form
+        window.location.href = "#";
         this.element.submit();
     }
```
Comment (blms) [inline sitemedia/js/controllers/search_controller.js:19]: @rlskoeser Can you test again to confirm it's working now, and is this an OK way to handle closing it? (I couldn't figure out how else to consistently close the modal without breaking the strict coupling of the anchor in the URL and visibility…)

**Review by rlskoeser — APPROVED** [review]
yes, that works — let's just expand the comment about the location change in the js code a bit so we remember why this is here later on when we're working on other parts of the search form

**Pull request merged**